pontoon/base/static/js/l20n.js

```diff
@@ -0,0 +1,2831 @@
+var L20n = (function () {
+  'use strict';
+
+  /*eslint no-magic-numbers: [0]*/
+
+  const locales2rules = {
+    'af': 3,
+    'ak': 4,
+    'am': 4,
+    'ar': 1,
+    'asa': 3,
+    'az': 0,
+    'be': 11,
+    'bem': 3,
+    'bez': 3,
+    'bg': 3,
+    'bh': 4,
+    'bm': 0,
+    'bn': 3,
+    'bo': 0,
+    'br': 20,
+    'brx': 3,
+    'bs': 11,
+    'ca': 3,
+    'cgg': 3,
+    'chr': 3,
+    'cs': 12,
+    'cy': 17,
+    'da': 3,
+    'de': 3,
+    'dv': 3,
+    'dz': 0,
+    'ee': 3,
+    'el': 3,
+    'en': 3,
+    'eo': 3,
+    'es': 3,
+    'et': 3,
+    'eu': 3,
+    'fa': 0,
+    'ff': 5,
+    'fi': 3,
+    'fil': 4,
+    'fo': 3,
+    'fr': 5,
+    'fur': 3,
+    'fy': 3,
+    'ga': 8,
+    'gd': 24,
+    'gl': 3,
+    'gsw': 3,
+    'gu': 3,
+    'guw': 4,
+    'gv': 23,
+    'ha': 3,
+    'haw': 3,
+    'he': 2,
+    'hi': 4,
+    'hr': 11,
+    'hu': 0,
+    'id': 0,
+    'ig': 0,
+    'ii': 0,
+    'is': 3,
+    'it': 3,
+    'iu': 7,
+    'ja': 0,
+    'jmc': 3,
+    'jv': 0,
+    'ka': 0,
+    'kab': 5,
+    'kaj': 3,
+    'kcg': 3,
+    'kde': 0,
+    'kea': 0,
+    'kk': 3,
+    'kl': 3,
+    'km': 0,
+    'kn': 0,
+    'ko': 0,
+    'ksb': 3,
+    'ksh': 21,
+    'ku': 3,
+    'kw': 7,
+    'lag': 18,
+    'lb': 3,
+    'lg': 3,
+    'ln': 4,
+    'lo': 0,
+    'lt': 10,
+    'lv': 6,
+    'mas': 3,
+    'mg': 4,
+    'mk': 16,
+    'ml': 3,
+    'mn': 3,
+    'mo': 9,
+    'mr': 3,
+    'ms': 0,
+    'mt': 15,
+    'my': 0,
+    'nah': 3,
+    'naq': 7,
+    'nb': 3,
+    'nd': 3,
+    'ne': 3,
+    'nl': 3,
+    'nn': 3,
+    'no': 3,
+    'nr': 3,
+    'nso': 4,
+    'ny': 3,
+    'nyn': 3,
+    'om': 3,
+    'or': 3,
+    'pa': 3,
+    'pap': 3,
+    'pl': 13,
+    'ps': 3,
+    'pt': 3,
+    'rm': 3,
+    'ro': 9,
+    'rof': 3,
+    'ru': 11,
+    'rwk': 3,
+    'sah': 0,
+    'saq': 3,
+    'se': 7,
+    'seh': 3,
+    'ses': 0,
+    'sg': 0,
+    'sh': 11,
+    'shi': 19,
+    'sk': 12,
+    'sl': 14,
+    'sma': 7,
+    'smi': 7,
+    'smj': 7,
+    'smn': 7,
+    'sms': 7,
+    'sn': 3,
+    'so': 3,
+    'sq': 3,
+    'sr': 11,
+    'ss': 3,
+    'ssy': 3,
+    'st': 3,
+    'sv': 3,
+    'sw': 3,
+    'syr': 3,
+    'ta': 3,
+    'te': 3,
+    'teo': 3,
+    'th': 0,
+    'ti': 4,
+    'tig': 3,
+    'tk': 3,
+    'tl': 4,
+    'tn': 3,
+    'to': 0,
+    'tr': 0,
+    'ts': 3,
+    'tzm': 22,
+    'uk': 11,
+    'ur': 3,
+    've': 3,
+    'vi': 0,
+    'vun': 3,
+    'wa': 4,
+    'wae': 3,
+    'wo': 0,
+    'xh': 3,
+    'xog': 3,
+    'yo': 0,
+    'zh': 0,
+    'zu': 3
+  };
+
+  // utility functions for plural rules methods
+  function isIn(n, list) {
+    return list.indexOf(n) !== -1;
+  }
+  function isBetween(n, start, end) {
+    return typeof n === typeof start && start <= n && n <= end;
+  }
+
+  // list of all plural rules methods:
+  // map an integer to the plural form name to use
+  const pluralRules = {
```

Comment:
We could move that map to the separate file and use it to solve https://bugzilla.mozilla.org/show_bug.cgi?id=1311053